README.md

```diff
@@ -1,5 +1,95 @@
 
-![alt text](assets/sign.png "Construction sign")
+## Decription
+
+NestedTensor is an abstraction of torch tensor which allows working with data of different sizes and length. 
+
+In general, there are two cases for which NestedTensors provide computational representations.
+
+**Lists of Tensors**
+Each Tensor constituent of the list it represents, if any, must be of its dtype, layout and device. The dimension of a constituent Tensor must be one less than the dimension of the NestedTensor. An empty list of Tensors yields a NestedTensor of dimension one.
+
+**Lists of NestedTensors**
+Each constituent NestedTensor must be of its dtype, layout and device. The dimension of a constituent NestedTensor must be one less than the dimension of the NestedTensor.
+
+## Prerequisites
+
+You will need this packages to be installed:
+
+- ipython
+- notebook
+- pytorch
+- torchvision
+
+If you have conda installed on your machine, you can install these via
+```
+conda install ipython pytorch notebook torchvision cpuonly -c pytorch-nightly
+```
+
+## Build 
+Run 
+```
+python setup.py develop
+```
+
+## Usage
+Import nested tensors and torch via ```from nestedtensor import torch```
+### Creation
+You can create a nester tensor by initializing its values direcrtly or by creating tensors first and then add them to nested tensor.
+
+```
+nt = torch.nested_tensor(
+    [
+        [
+            torch.rand(2, 3),
+            torch.rand(4, 5)
+        ],
+        [
+            torch.rand(1, 2)
+        ]
+    ])
+```
+
+```
+a = torch.tensor([1])
+b = torch.tensor([[2, 2],
+                  [3, 3],
+                  [4, 4],
+                  [5, 5]])
+nt2 = torch.nested_tensor([[a],[b]])
+```
+
+The level of nesting is inferred from the input. The constructor always copies. Whatever you pass into the constructor will share no data with what the constructor returns. This matches torch.tensor's behavior.
+
+If given a NestedTensor or Tensor it will return a detached copy, which is consistent with the behavior of torch.tensor. Remember that you cannot mix Tensors and NestedTensors within a given list.
+
+### Conversion/unbind()
+A user can retrieve the constituent Tensors via unbind. Unbind is currently used by torch to turn Tensors into tuples of Tensors. Unbind always returns a tuple of views. We do not (yet support a dimension argument to unbind for NestedTensors, because it forces us to argue about shape.
+
+```
+a = [ \
+       [torch.rand(2, 3), torch.rand(4, 5)], \
+       [torch.rand(6, 7)] \
+      ]
+
+b = torch.nested_tensor(a)
+b1 = b.unbind() # Tuple of 2 NestedTensors
+b2 = b1[0].unbind() # Tuple of 2 Tensors
+```
+
+### Other Ops
+For all other ops, please, see examples.
```

Comment:
There's a codegen function that gives you a list of all bindings, but I'm waiting on ```__torch_function__``` to give us an easy way of showing support. Really we shouldn't use monkeypatching. Here is the branch to use torch function: https://github.com/cpuhrsch/nestedtensor/tree/torchfunction

Reply:
I think that if its a branch, we should not put it into a doc. its like telling people that pytroch has nested tensors if you pick the right branch.

Reply:
Yes, this was just for your information.